discord/ext/commands/converter.py

```diff
@@ -159,6 +159,80 @@ class UserConverter(IDConverter):
 
         return result
 
+class MessageConverter(IDConverter):
+    """Converts to a :class:`discord.Message`.
+
+    The lookup strategu is as follows (in order):
+
+    1. Lookup by "{channel ID}-{message ID}" (got by shift-clicking on "Copy ID")
+    2. Lookup by message URI
+    3. Lookup by message ID (the message **must** be in the context channel)
+    """
+
+    async def convert(self, ctx, argument: str) -> discord.Message:
+        """|coro|
+
+        The method to override to do conversion logic.
+
+        If an error is found while converting, it is recommended to
+        raise a :exc:`.CommandError` derived exception as it will
+        properly propagate to the error handlers.
+
+        Parameters
+        -----------
+        ctx: :class:`.Context`
+            The invocation context that the argument is being used in.
+        argument: :class:`str`
+            The argument that is being converted.
+        """
+        if len(argument.split()) > 1:
+            raise BadArgument("The message must be an ID or an URI.")
+        exception = BadArgument('Message "{msg}" not found.'.format(msg=argument))
+        if "-" in argument:
+            channel_id, message_id = [self._get_id_match(x) for x in argument.split("-")]
+            if not channel_id or not message_id:
+                raise exception
+            channel_id = int(channel_id.group(1))
+            message_id = int(message_id.group(1))
+            channel = ctx.bot.get_channel(channel_id)
+            if not channel:
+                raise exception
+            try:
+                message = await channel.fetch_message(message_id)
+            except discord.errors.NotFound:
+                raise exception
+            except discord.errors.Forbidden:
+                raise BadArgument("Can't read messages in {channel}".format(channel=channel.mention))
+        elif "://discordapp.com/channels/" in argument:
```

Comment:
Once again this is better suited with a regex rather than this approach. This also does not work for `ptb` and `canary` links.